Source/Core/VideoBackends/OGL/ProgramShaderCache.cpp

```diff
@@ -99,11 +99,38 @@ void SHADER::SetProgramVariables()
   }
 }
 
+bool IsDualSourceBlendActive()
+{
+  if (!g_ActiveConfig.backend_info.bSupportsDualSourceBlend)
+    return false;
+
+  GLboolean blendEnabled;
+  glGetBooleanv(GL_BLEND, &blendEnabled);
+  if (!blendEnabled) return false;
+
+  GLint blendSrc;
+  GLint blendSrcAlpha;
+  GLint blendDest;
+  GLint blendDestAlpha;
+  glGetIntegerv(GL_BLEND_SRC_RGB, &blendSrc);
+  glGetIntegerv(GL_BLEND_DST_RGB, &blendDest);
+  glGetIntegerv(GL_BLEND_SRC_ALPHA, &blendSrcAlpha);
+  glGetIntegerv(GL_BLEND_DST_ALPHA, &blendDestAlpha);
```

Comment:
If possible I'd prefer never to disable dual-source blending. But when the blender doesn't actually use the alpha value anywhere, then dual-source blending is disabled regardless.

I could make this part of the shader UID, but then we'd have to compile a bunch of extra shaders. Though if we're going to use this PR we'd need to re-link the shader every time, because program bindings only apply the next time the program is linked.